Resources/views/economy/show.html.twig

```diff
@@ -33,34 +33,132 @@
     {{ parent() }}
 
     <script type="text/javascript">
         $(document).ready(function () {
-            $('.js-itkdev-monthpicker').monthpicker($('.js-monthpicker-from'), $('.js-monthpicker-to'), {
-                headline: "{{ 'economy_entry.form.service.monthpicker_headline'|trans }}",
-                applyButtonText: "{{ 'economy_entry.form.service.monthpicker_applybutton_text'|trans }}",
-                cancelButtonText: "{{ 'economy_entry.form.service.monthpicker_cancelbutton_text'|trans }}"
-            });
 
             $('#economy_entry_type').change(function () {
                 $(this).closest('form').submit();
                 $('#base_economy_entry').hide();
-                $('.service-economy-entry-monthpicker').hide();
                 $('#service_economy_entry').hide();
                 $('#js-economy-form-spinner').show();
             });
 
             // Apply datetimepicker to all js-datepicker elements.
             $('.js-datepicker').datetimepicker();
+
+            $('#service_economy_entry_datepicker').click(function() {
+
+                $('#service_economy_entry_datepicker_modal').modal();
+            });
+
+            $('#income_economy_entry_datepicker').click(function() {
+
+                $('#income_economy_entry_datepicker_modal').modal();
+            });
+
+            $('.datepicker-period').datetimepicker({
+                inline: true,
+                format: 'L',
+                useCurrent: false,
+                defaultDate: false,
+            });
+
+            $('#datepicker_from').on('change.datetimepicker', function(e) {
+                $('#datepicker_to').datetimepicker('minDate', e.date);
+            });
+
+            $('#datepicker_to').on('change.datetimepicker', function(e) {
+                $('#datepicker_from').datetimepicker('maxDate', e.date);
+            });
+
+            $('#datepicker_period_save').click(function() {
+
+                if ($('#service_economy_entry_periodFrom').val()) {
+                    let from = $('#service_economy_entry_periodFrom').val();
+                    let to = $('#service_economy_entry_periodTo').val();
+
+                    $('#service_economy_entry_datepicker').text(from + ' - ' + to);
+
+                    $('#service_economy_entry_datepicker_modal').modal('toggle');
+                }
+
+                if ($('#income_economy_entry_periodFrom').val()) {
+                    let from = $('#income_economy_entry_periodFrom').val();
+                    let to = $('#income_economy_entry_periodTo').val();
+
+                    $('#income_economy_entry_datepicker').text(from + ' - ' + to);
+
+                    $('#income_economy_entry_datepicker_modal').modal('toggle');
+                }
+            });
+
+            let futureSavingsInput;
+            let futureSavingsModalId;
+            let repaymentInput;
+            let repaymentModalId;
+
+            $('.future-savings-datepicker').click(function() {
```

Comment:
Replace all the modals with one where the js reinitializes the datepicker targets for each modal open action.